go/datas/serialize_hints.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package datas
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+
+	"github.com/attic-labs/noms/go/chunks"
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/hash"
+)
+
+func serializeHashes(w io.Writer, batch chunks.ReadBatch) {
+	err := binary.Write(w, binary.BigEndian, uint32(len(batch))) // 4 billion hashes is probably absurd. Maybe this should be smaller?
+	d.PanicIfError(err)
+	for h := range batch {
+		serializeHash(w, h)
+	}
+}
+
+func serializeHash(w io.Writer, h hash.Hash) {
+	n, err := io.Copy(w, bytes.NewReader(h[:]))
```

Comment:
i think it's safe here to just `w.Write(h[:])`. From the `Writer` go docs: "Write must return a non-nil error if it returns n < len(p)."